src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,834 @@
+                loaderAssemblyName: typeof(DateTimeTransformer).Assembly.FullName);
+        }
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // name of input column
+            // column prefix
+            // byte length of columns to drop array
+            // byte array of columns to drop
+            // length of C++ state array
+            // C++ byte state array
+
+            ctx.Writer.Write(_column.Source);
+            ctx.Writer.Write(_column.Prefix);
+
+            ctx.Writer.Write(_columnsToDrop == null ? 0 : _columnsToDrop.Length);
+            if (_columnsToDrop != null)
+            {
+                foreach (var toDrop in _columnsToDrop)
+                    ctx.Writer.Write((byte)toDrop);
+            }
+
+            var data = _column.CreateTransformerSaveData();
+            ctx.Writer.Write(data.Length);
+            ctx.Writer.Write(data);
+        }
+
+        public void Dispose()
+        {
+            _column.Dispose();
+        }
+
+        #region C++ Safe handle classes
+
+        internal class TransformedDataSafeHandle : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private readonly DestroyTransformedDataNative _destroyTransformedDataHandler;
+
+            public TransformedDataSafeHandle(IntPtr handle, DestroyTransformedDataNative destroyTransformedDataHandler) : base(true)
+            {
+                SetHandle(handle);
+                _destroyTransformedDataHandler = destroyTransformedDataHandler;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return _destroyTransformedDataHandler(handle, out IntPtr errorHandle);
+            }
+        }
+
+        #endregion
+
+        #region TimePoint
+
+        [StructLayoutAttribute(LayoutKind.Sequential)]
+        internal struct TimePoint
+        {
+            public int Year;
+            public byte Month;
+            public byte Day;
+            public byte Hour;
+            public byte Minute;
+            public byte Second;
+            public byte AmPm;
+            public byte Hour12;
+            public byte DayOfWeek;
+            public byte DayOfQuarter;
+            public ushort DayOfYear;
+            public ushort WeekOfMonth;
+            public byte QuarterOfYear;
+            public byte HalfOfYear;
+            public byte WeekIso;
+            public int YearIso;
+            public string MonthLabel;
+            public string AmPmLabel;
+            public string DayOfWeekLabel;
+            public string HolidayName;
+            public byte IsPaidTimeOff;
+
+            internal unsafe TimePoint(byte* rawData)
```

Comment:
(suggestion) you could make this safe code by using `ReadOnlySpan<byte>` instead of `byte*`.

Reply:
I will make that change. How does the performance compare between the raw pointer and the span?

Reply:
There should be no measurable performance difference between them.

Reply:
There should be no measurable performance difference between them.

Reply:
I have changed it to use spans. Because of a double pointer issue I still need unsafe, but now the constructor is safe and just internal things are unsafe.